src/main/java/org/wso2/carbon/inbound/poll/dbeventlistener/DBEventPollingConsumer.java

```diff
@@ -0,0 +1,380 @@
+import org.wso2.carbon.base.MultitenantConstants;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.inbound.endpoint.protocol.generic.GenericPollingConsumer;
+
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.List;
+import java.util.Properties;
+
+import java.sql.DatabaseMetaData;
+import java.sql.SQLException;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.Statement;
+import java.sql.DriverManager;
+import java.sql.Types;
+
+public class DBEventPollingConsumer extends GenericPollingConsumer {
+
+    private static final Log log = LogFactory.getLog(DBEventPollingConsumer.class);
+    
+    private String driverClass;
+    private String dbURL;
+    private String dbUsername;
+    private String dbPassword;
+    private String filteringColumnName;
+    private String filteringCriteria;
+    private Connection connection = null;
+    private MessageContext msgCtx;
+    private String deleteQuery = null;
+    private String updateQuery = null;
+    private String lastProcessedTimestamp = null;
+    private String registryPath = null;
+    private String inboundName = null;
+    private String[] primaryKeysFromConfig = null;
+    private String connectionValidationQuery = null;
+
+    /**
+     * @param properties
+     * @param name
+     * @param synapseEnvironment
+     * @param scanInterval
+     * @param injectingSeq
+     * @param onErrorSeq
+     * @param coordination
+     * @param sequential
+     */
+    public DBEventPollingConsumer(Properties properties, String name, SynapseEnvironment synapseEnvironment,
+            long scanInterval, String injectingSeq, String onErrorSeq, boolean coordination, boolean sequential) {
+        super(properties, name, synapseEnvironment, scanInterval, injectingSeq, onErrorSeq, coordination, sequential);
+        driverClass = properties.getProperty(DBEventConstants.DB_DRIVER);
+        dbURL = properties.getProperty(DBEventConstants.DB_URL);
+        dbUsername = properties.getProperty(DBEventConstants.DB_USERNAME);
+        dbPassword = properties.getProperty(DBEventConstants.DB_PASSWORD);
+        filteringCriteria = properties.getProperty(DBEventConstants.DB_FILTERING_CRITERIA);
+        filteringColumnName = properties.getProperty(DBEventConstants.DB_FILTERING_COLUMN_NAME);
+        registryPath = properties.getProperty(DBEventConstants.REGISTRY_PATH);
+        primaryKeysFromConfig = properties.getProperty(DBEventConstants.TABLE_PRIMARY_KEYS).split(",");
+        connectionValidationQuery = properties.getProperty(DBEventConstants.CONNECTION_VALIDATION_QUERY);
+        if(StringUtils.isEmpty(registryPath)) {
+            registryPath = name;
+        }
+        inboundName = name;
+    }
+
+    /**
+     * Inject the message to the sequence
+     *
+     * @param object
+     * @return status
+     */
+    public boolean inject(OMElement object) {
```

Comment:
Why this is a public method. This is only been called within the class? If so make it private.